numpy/core/src/umath/loops.c.src

```diff
@@ -540,6 +540,24 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 @ATTR@ void
         BINARY_LOOP_FAST(@type@, @type@, *out = in1 @OP@ in2);
     }
 }
+
+NPY_NO_EXPORT NPY_GCC_OPT_3 @ATTR@ int
+@TYPE@_@kind@@isa@_indexed(PyArrayMethod_Context *NPY_UNUSED(context), char * const*args, npy_intp const *dimensions, npy_intp const *steps, NpyAuxData *NPY_UNUSED(func))
+{
+    char *ip1 = args[0];
+    npy_intp *indx = (npy_intp *)args[1];
+    char *value = args[2];
+    npy_intp is1 = steps[0], is2 = steps[1] / sizeof(npy_intp), os1 = steps[2];
+    npy_intp n = dimensions[0];
+    npy_intp i;
+    @type@ *indexed;
+    for(i = 0; i < n; i++, indx += is2, value += os1) {
+        indexed = (@type@ *)(ip1 + is1 * indx[0]);
```

Comment:
So, I *think* I'd prefer `indexed = (@type@ *)(ip1 + is1 * (npy_intp *)indx))` here. But another nitpick, really.

Reply:
ok

Reply:
On second thought, I really prefer the `indx[0]` syntax: I find it easier to parse and it is ~8 characters shorter. Anywhere I can avoid a forced cast seems to be a win for strong typing.

Reply:
I.e. this annoying and hard to get, but it seems to me that this may be failing (assuming that intp alignment is 4 and not 8 and we are on 64bits):
```
indx = np.arange(5).astype(dtype="i4,intp")["f1"]
```
on my computer `indx.flags` is not aligned though...

I am OK with assuming unit steps (non-unit steps are very uncommon after all and it might be a bit faster), would be good to add a check, though.

Reply:
Interesting thoughts.  Indeed, I don't very much like the casts either, but all the other loops are written like that... Anyway, seems like this has converged.